### PR TITLE
Wait for worker MCP to be updated before running RHEL worker upgrade

### PR DESCRIPTION
**Pull request by gpei**
In the previous CPOU upgrade workflow, after unpausing the worker MachineConfigPool (MCP), the process skipped waiting for worker MCP rolling updates if RHEL nodes were detected. This caused **RHEL nodes to retain the original node MachineConfig after RPM package upgrades**, resulting in the new kubelet version being configured with outdated launch parameters. Consequently, kubelet fails to start.

Example failure: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/65410/rehearse-65410-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-cpou-upgrade-from-4.12-azure-ipi-workers-rhel8-f28/1927589971233869824

When updating the RHEL worker from 4.12 to 4.14, the kubelet service failed to be started as following:

```
May 29 21:18:30 ci-op-wx29c5np-9c8a2-jj2qq-rhel-1 systemd[1]: Starting Kubernetes Kubelet...
May 29 21:18:30 ci-op-wx29c5np-9c8a2-jj2qq-rhel-1 kubenswrapper[20864]: E0529 21:18:30.754260   20864 run.go:74] "command failed" err="failed to parse kubelet flag: unknown flag: --container-runtime"
May 29 21:18:30 ci-op-wx29c5np-9c8a2-jj2qq-rhel-1 systemd[1]: kubelet.service: Main process exited, code=exited, status=1/FAILURE
May 29 21:18:30 ci-op-wx29c5np-9c8a2-jj2qq-rhel-1 systemd[1]: kubelet.service: Failed with result 'exit-code'.
May 29 21:18:30 ci-op-wx29c5np-9c8a2-jj2qq-rhel-1 systemd[1]: Failed to start Kubernetes Kubelet.
```
The kubelet service start parameters on the problem RHEL node, coming from 4.12 node MachineConfig.
```
ExecStart=/usr/local/bin/kubenswrapper \
    /usr/bin/kubelet \
      --config=/etc/kubernetes/kubelet.conf \
      --bootstrap-kubeconfig=/etc/kubernetes/kubeconfig \
      --kubeconfig=/var/lib/kubelet/kubeconfig \
      --container-runtime=remote \
      --container-runtime-endpoint=/var/run/crio/crio.sock \
...
```
The kubelet service log also told the flag `--container-runtime` is a deprecated one and would be removed in k8s 1.27(OCP-4.14)
```
May 29 13:20:17 ci-op-wx29c5np-9c8a2-jj2qq-rhel-1 kubenswrapper[4656]: Flag --container-runtime has been deprecated, will be removed in 1.27 as the only valid value is 'remote'
```
In fresh installed 4.13 and 4.14 clusters, the kubelet service was configured **without** this Flag.
```
ExecStart=/usr/local/bin/kubenswrapper \
    /usr/bin/kubelet \
      --config=/etc/kubernetes/kubelet.conf \
      --bootstrap-kubeconfig=/etc/kubernetes/kubeconfig \
      --kubeconfig=/var/lib/kubelet/kubeconfig \
      --container-runtime-endpoint=/var/run/crio/crio.sock \
      --runtime-cgroups=/system.slice/crio.service \
```

So we need to run the ansible playbook to update RHEL nodes **only after the new MachineConfig is fully deployed to all nodes**, to ensure configuration-version consistency. 

P.S. For scenarios requiring immediate RHEL node upgrades without waiting for MCP updates (e.g., specific version upgrade test jobs to workaround MCO bugs), we still could use the existing `UPGRADE_RHEL_WORKER_BEFOREHAND` environment variable to bypass this workflow for potential edge cases in the future.

P.P.S. Added a workaround during 4.14 to 4.16 cluster upgrade to install the required cloud provider packages on the RHEL nodes, so that the unpausing worker mcp step could be executed successfully.

### CODE REVIEW

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-cpou-upgrade-from-4.12-azure-ipi-workers-rhel8-f28

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-cpou-upgrade-from-4.12-azure-ipi-workers-rhel8-f28

**Comment by jianlinliu** [comment]
> For scenarios requiring immediate RHEL node upgrades without waiting for MCP updates (e.g., specific version upgrade test jobs to workaround MCO bugs), we still could use the existing UPGRADE_RHEL_WORKER_BEFOREHAND environment variable to bypass this workflow for potential edge cases in the future.

From `openshift-upgrade-qe-test-cpou` chain:
```
  - ref: cucushift-upgrade-cpou-pause-worker-mcp
  - ref: cucushift-chainupgrade-toimage
  - ref: cucushift-upgrade-cpou-unpause-worker-mcp
  - ref: cucushift-upgrade-rhel-worker
```

Before https://github.com/openshift/release/pull/65410/files#diff-440cd1be9ac67cd04d044f301ea04d31142584464112b57a6273baf49d6920b3L100, sounds like it need to skip `check_mcp` on purpose. while in this PR, it has to run `check_mcp`. Per the above steps in the cpou chain, once that, I guess the process would fail on `check_mcp` in `cucushift-upgrade-cpou-unpause-worker-mcp`, it would never come to `cucushift-upgrade-rhel-worker`.

@gpei do you mind to add `azure-ipi-workers-rhel8-f28` to openshift-openshift-tests-private-release-4.16__amd64-nightly-4.16-cpou-upgrade-from-4.14.yaml and openshift-openshift-tests-private-release-4.18__amd64-nightly-4.18-cpou-upgrade-from-4.16.yaml for double confirm ?

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-4.16-cpou-upgrade-from-4.14-azure-ipi-workers-rhel8-f28

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-4.16-cpou-upgrade-from-4.14-azure-ipi-workers-rhel8-f28 periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-cpou-upgrade-from-4.12-azure-ipi-workers-rhel8-f28

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-4.16-cpou-upgrade-from-4.14-azure-ipi-workers-rhel8-f28

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-4.16-cpou-upgrade-from-4.14-azure-ipi-workers-rhel8-f28

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-4.16-cpou-upgrade-from-4.14-azure-ipi-workers-rhel8-f28

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-4.16-cpou-upgrade-from-4.14-azure-ipi-workers-rhel8-f28

File: ci-operator/step-registry/cucushift/upgrade/cpou/unpause-worker-mcp/cucushift-upgrade-cpou-unpause-worker-mcp-commands.sh [hunk 1]
```diff
@@ -100,11 +100,5 @@ do
...
-    if [[ "${current_minor_ver}" -lt "19" ]] && [[ $(oc get nodes -l node.openshift.io/os_id=rhel) != "" ]]; then
-        echo "Found rhel worker, this step is supposed to be used in eus upgrade, skipping mcp checking here, need to check it after rhel worker upgraded..."
-        run_command "oc get machineconfigpools"
-        run_command "oc get node -owide"
-    else
-        check_mcp ${mcp} "True"
-    fi
-done
\ No newline at end of file
+    check_mcp ${mcp} "True"
```
Comment (jianlinliu) [inline ci-operator/step-registry/cucushift/upgrade/cpou/unpause-worker-mcp/cucushift-upgrade-cpou-unpause-worker-mcp-commands.sh:101]: Does that mean worker mcp always can be updated even if `cucushift-upgrade-rhel-worker` is not executed yet.

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-4.18-cpou-upgrade-from-4.16-azure-ipi-workers-rhel8-f28

**Comment by gpei** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-4.18-cpou-upgrade-from-4.16-azure-ipi-workers-rhel8-f28

**Comment by gpei** [comment]
@jianlinliu @jiajliu With the current workflow in this PR, all the following test got passed now, please help to review again, thanks.

- openshift-tests-private-release-4.14-amd64-nightly-4.14-cpou-upgrade-from-4.12-azure-ipi-workers-rhel8-f28: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/65478/rehearse-65478-periodic-ci-openshift-openshift-tests-private-release-4.14-amd64-nightly-4.14-cpou-upgrade-from-4.12-azure-ipi-workers-rhel8-f28/1929850293520961536
- openshift-tests-private-release-4.16-amd64-nightly-4.16-cpou-upgrade-from-4.14-azure-ipi-workers-rhel8-f28(With the temporary workaround applied): https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/65478/rehearse-65478-periodic-ci-openshift-openshift-tests-private-release-4.16-amd64-nightly-4.16-cpou-upgrade-from-4.14-azure-ipi-workers-rhel8-f28/1930486474004762624
- openshift-tests-private-release-4.18-amd64-nightly-4.18-cpou-upgrade-from-4.16-azure-ipi-workers-rhel8-f28: https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/65478/rehearse-65478-periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-4.18-cpou-upgrade-from-4.16-azure-ipi-workers-rhel8-f28/1930758764638507008

File: ci-operator/step-registry/cucushift/upgrade/cpou/unpause-worker-mcp/cucushift-upgrade-cpou-unpause-worker-mcp-commands.sh [hunk 2]
```diff
@@ -100,11 +100,5 @@ do
     printf "\n"
     unpause ${mcp}
     printf "\n"
-    if [[ "${current_minor_ver}" -lt "19" ]] && [[ $(oc get nodes -l node.openshift.io/os_id=rhel) != "" ]]; then
```
Comment (jiajliu) [inline ci-operator/step-registry/cucushift/upgrade/cpou/unpause-worker-mcp/cucushift-upgrade-cpou-unpause-worker-mcp-commands.sh:103]: @gpei could you help remove above line93-line94 together? It was added for checking current_minor_ver in previous pr.

Comment (gpei) [reply ci-operator/step-registry/cucushift/upgrade/cpou/unpause-worker-mcp/cucushift-upgrade-cpou-unpause-worker-mcp-commands.sh:103]: updated
